db/python/layers/ourdna/dashboard.py

```diff
@@ -15,93 +15,106 @@
-            sample=sample, property_name='collection-time'
-        )
-        _process_end_time = self.get_meta_property(
-            sample=sample, property_name='process-end-time'
-        )
-        if _collection_time is None or _process_end_time is None:
+    def get_time_difference_in_seconds(
+        self, start_property: str, end_property: str
+    ) -> int | None:
+        """Calculate time difference in seconds between two meta properties."""
+        start_time = self.get_property(start_property)
+        end_time = self.get_property(end_property)
+        if start_time is None or end_time is None:
             return None
-
         time_taken = datetime.strptime(
-            _process_end_time, '%Y-%m-%d %H:%M:%S'
-        ) - datetime.strptime(_collection_time, '%Y-%m-%d %H:%M:%S')
-
+            end_time, '%Y-%m-%d %H:%M:%S'
+        ) - datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S')
         return int(time_taken.total_seconds())
 
-    def get_processing_times_by_site(
-        self, sample: Sample
-    ) -> tuple[str | None, int | None]:
-        """
-        I want to know what the sample processing times were for samples at each designated site (BBV, Garvan, Westmead, etc)
-        @fields: process-start-time, process-end-time, processing-site where the time fields are of the format '2022-07-03 13:28:00'
-        """
-        _process_start_time = self.get_meta_property(
-            sample=sample, property_name='process-start-time'
-        )
-        _process_end_time = self.get_meta_property(
-            sample=sample, property_name='process-end-time'
-        )
-        _processing_site = self.get_meta_property(
-            sample=sample, property_name='processing-site'
-        )
-        if (
-            _process_start_time is None
-            or _process_end_time is None
-            or _processing_site is None
-        ):
-            return None, None
-
-        processing_time = datetime.strptime(
-            _process_end_time, '%Y-%m-%d %H:%M:%S'
-        ) - datetime.strptime(_process_start_time, '%Y-%m-%d %H:%M:%S')
-
-        return _processing_site, int(processing_time.total_seconds())
-
-    def get_collection_to_process_start_time(self, sample: Sample) -> int | None:
-        """
-        I want to know how long it has been since the sample was collected - SAMPLE TABLE
-        @fields: collection-time, process-start-time
-        """
-        _collection_time = self.get_meta_property(
-            sample=sample, property_name='collection-time'
-        )
-        _process_start_time = self.get_meta_property(
-            sample=sample, property_name='process-start-time'
-        )
-        if _collection_time is None or _process_start_time is None:
-            return None
+    @staticmethod
+    def from_sample(sample: Sample) -> 'SampleProcessMeta':
+        """Create a SampleProcessMeta instance from a Sample instance."""
+        instance = SampleProcessMeta()
```

Comment:
EG: it should be as simple as this
```suggestion
        instance = SampleProcessMeta(meta=sample.meta)
```

Or just change this whole thing to an init:

```python
def __init__(self, sample: Sample):
    self.meta = sample.meta
```